src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -139,47 +139,80 @@ private static int atLeast2(int value, String str)
     public static String printScalingParameter(int w)
     {
         if (w < 0)
-            return "L" + Integer.toString(2 - w);
+            return 'L' + Integer.toString(2 - w);
         else if (w > 0)
-            return "T" + Integer.toString(w + 2);
+            return 'T' + Integer.toString(w + 2);
         else
             return "N";
     }
 
+    private TimeUUID nextTimeUUID()
+    {
+        return TimeUUID.Generator.nextTimeUUID().withSequence(0);
```

Comment:
Reading further, I realize this is due to how `CompositeLifecycleTransaction` name its subtasks. It's fine, but again, worth documenting since the underlying reason is a bit "distant".